lib/modules/dosomething/dosomething_signup/dosomething_signup.module

```diff
@@ -308,22 +307,42 @@ function dosomething_signup_exists($nid, $runId = NULL, $uid = NULL, $presignup
       ->execute();
   }
   else {
+    // First check to see if a user has a signup with a run.
     $result = db_select('dosomething_signup', 's')
       ->condition('uid', $uid)
       ->condition('nid', $nid)
-      ->condition('run_nid', $runId)
+      ->condition('run_nid', $run_nid)
       ->fields('s', array('sid'))
       ->execute();
   }
+  $sid = $result->fetchField();
 
-  if (!isset($result)) {
-    return FALSE;
+  // If not see if they are signed up for the campaign without the run.
+  // This section is added to test for duplicate signup problems.
+  // Refs #6095.
+  if (!is_numeric($sid)) {
```

Comment:
This will also trigger if we're running in "presignup" mode, and that query returned FALSE-y. There may be no side effects (not sure I have all the context), but you could eliminate entirely by checking for `&& !$presignup` here.